ci-operator/step-registry/gcp/provision/private-dns-zone-in-host-project/gcp-provision-private-dns-zone-in-host-project-commands.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+echo "Reading variables from ${SHARED_DIR}/xpn.json..."
+NETWORK=$(jq -r '.clusterNetwork' "${SHARED_DIR}/xpn.json")
+
+CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+
+if [[ -s "${SHARED_DIR}/xpn.json" ]] && [[ -f "${CLUSTER_PROFILE_DIR}/xpn_creds.json" ]]; then
+  echo "Activating XPN service-account..."
+  GOOGLE_CLOUD_XPN_KEYFILE_JSON="${CLUSTER_PROFILE_DIR}/xpn_creds.json"
+  gcloud auth activate-service-account --key-file="${GOOGLE_CLOUD_XPN_KEYFILE_JSON}"
+fi
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  GCP_PROJECT="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GCP_PROJECT}"
+fi
```

Comment:
We do not need to login service project in this step, right?

Reply:
It'll login if not already. So?

Reply:
If the step does not need it, pls do not code it there to keep tidy.

Reply:
By build [log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/33928/rehearse-33928-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-gcp/1592129494838153216/artifacts/installer-rehearse-gcp/gcp-provision-private-dns-zone-in-service-project/build-log.txt), the login is needed.